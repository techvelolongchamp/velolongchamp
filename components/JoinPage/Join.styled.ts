import styled from 'styled-components'

export const FrameContainer = styled.div`
  background: url(..//loader.gif) center center no-repeat;
`

export const SubTitle = styled.p`
  font-size: ${({ theme }) => theme.fonts.large};
  margin: ${({ theme }) => theme.spacing.l} 0;
  color: ${({ theme }) => theme.colors.brandTertiary};
  font-weight: 600;
`

export const Message = styled.div`
  text-align: center;
  a {
    cursor: pointer;
    color: inherit;
    &:hover {
      text-decoration: underline;
    }
  }
`
