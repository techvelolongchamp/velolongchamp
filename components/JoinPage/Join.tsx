import React from 'react'
import { useIntl } from 'react-intl'

import { SubTitle, FrameContainer, Message } from './Join.styled'

const Join = () => {
  const { formatMessage } = useIntl()
  return (
    <>
      <SubTitle>{formatMessage({ id: 'join.subtitle' })}</SubTitle>
      <FrameContainer>
        <iframe
          title={formatMessage({ id: 'join.title' })}
          id="haWidget"
          data-hj-allow-iframe=""
          scrolling="auto"
          src="https://www.helloasso.com/associations/velo-longchamp/adhesions/adhesion-association-velo-longchamp/widget"
          style={{ width: '100%', height: '750px', border: 'none' }}
        ></iframe>
      </FrameContainer>
      <Message>
        {formatMessage(
          { id: 'join.problem' },
          {
            a: (chunk: React.ReactNode) => (
              <a
                href="https://www.helloasso.com/associations/velo-longchamp/adhesions/adhesion-association-velo-longchamp"
                target="_blank"
                rel="noopener noreferrer"
                title="HelloAsso"
              >
                {chunk}
              </a>
            ),
          }
        )}
      </Message>
    </>
  )
}

export default Join
