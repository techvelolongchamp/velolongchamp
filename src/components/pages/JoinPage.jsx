import React from 'react'
import styled from 'styled-components'

import Layout from '../Layout'
import Header from '../Header'
import Head from '../Head'
import ThirdarySection from '../sections/ThirdarySection'

const FrameContainer = styled.div`
  background: url(..//loader.gif) center center no-repeat;
`

const SubTitle = styled.p`
  font-size: ${({ theme }) => theme.fonts.large};
  margin: ${({ theme }) => theme.spacing.l} 0;
  color: ${({ theme }) => theme.colors.brandTertiary};
  font-weight: 600;
`

const Message = styled.div`
  text-align: center;
  a {
    cursor: pointer;
    color: inherit;
    &:hover {
      text-decoration: underline;
    }
  }
`

const JoinFrameI18n = {
  fr: {
    join: 'Adhérer gratuitement à Vélo Longchamp :',
    problem:
      'Si vous rencontrez un problème avec le formulaire, vous pouvez y accéder directement depuis le site de HelloAsso en cliquant ',
    link: 'ici',
  },
  en: {
    join: 'Join Vélo Longchamp for free:',
    problem:
      'If you come across a problem with the form, you can directly access it on HelloAsso website by clicking ',
    link: 'here',
  },
}

const JoinFrame = ({ lng }) => {
  const trad = JoinFrameI18n[lng]
  return (
    <Layout lng={lng}>
      <Header noScroll lng={lng} />
      <Head title="Blog" url="/blog" lng={lng} />
      <ThirdarySection title="Blog" useH1>
        <SubTitle>{trad.join}</SubTitle>
        <FrameContainer>
          <iframe
            id="haWidget"
            allowtransparency="true"
            data-hj-allow-iframe=""
            scrolling="auto"
            src="https://www.helloasso.com/associations/velo-longchamp/adhesions/adherer-gratuitement-a-velo-longchamp-pour-la-renovation-de-l-anneau-cycliste/widget"
            style={{ width: '100%', height: '750px', border: 'none' }}
          ></iframe>
        </FrameContainer>
        <Message>
          {trad.problem}
          <a
            href="https://www.helloasso.com/associations/velo-longchamp/adhesions/adherer-gratuitement-a-velo-longchamp-pour-la-renovation-de-l-anneau-cycliste"
            target="_blank"
            rel="noopener noreferrer"
            title="HelloAsso"
          >
            {trad.link}
          </a>
        </Message>
      </ThirdarySection>
    </Layout>
  )
}

export default JoinFrame
