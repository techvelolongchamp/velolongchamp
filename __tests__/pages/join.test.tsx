import React from 'react'
import { RenderWithProvider as render } from '../provider'

import JoinPage from '../../pages/join'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('JoinPage unit tests', () => {
  it('should render correctly', () => {
    window.matchMedia = jest.fn().mockImplementation(() => ({
      addListener: jest.fn(),
      removeListener: jest.fn(),
      matches: false,
    }))
    useRouter.mockImplementation(() => ({ locale: 'fr', pathname: '/' }))
    const { container } = render(<JoinPage />)

    expect(container).toMatchSnapshot()
  })
})
