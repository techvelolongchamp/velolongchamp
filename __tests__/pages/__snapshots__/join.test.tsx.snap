// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinPage unit tests should render correctly 1`] = `
.c22 {
  background: #f7f7F7;
  border-top: 1px solid #f7f7F7;
  color: #999999;
  padding: 10px 0 5px;
  text-align: center;
}

.c22 a {
  color: #999999;
}

.c22 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c23 {
  background: #0d171b;
  bottom: 0;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  color: white;
  font-size: 14px;
  padding: 10px;
  position: fixed;
  text-align: center;
  width: 100vw;
  z-index: 1000;
}

.c23 div {
  margin-top: 10px;
}

.c23 a {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  padding: 5px;
}

.c23 button {
  background: #419a1c;
  border-radius: 5px;
  color: #fff;
  margin: 0 10px;
  outline: done;
  padding: 5px;
  width: 40px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-top: 80px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: darkorange;
  border-bottom-left-radius: 30px;
  border-top-left-radius: 30px;
  color: white;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 40px;
  font-weight: 100;
  position: fixed;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  right: 0;
  top: 120px;
  z-index: 10;
}

.c1 svg {
  margin: 10px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: darkorange;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  right: -200px;
  top: 120px;
  z-index: 10;
  width: 200px;
  font-size: 16px;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  border-bottom-left-radius: 10px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 10px 20px 0;
  padding: 5px;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #419a1c;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 120px;
  border-radius: 5px;
}

.c3:hover {
  color: #f7f7F7;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c15 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c15:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 5;
  background: #f7f7F7;
  padding: 5px 10px;
  border-bottom: 1px solid #fff;
  color: #0d171b;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  margin: 0;
}

.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-left: 1px solid #6d6e71;
  color: #999999;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16px;
  height: 30px;
  margin: 0 20px;
  padding-left: 20px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12:hover {
  color: #666666;
}

.c13 {
  background: #6d6e71;
  border-radius: 5px;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 10px;
}

.c9 {
  height: 70px;
  padding: 5px 0;
}

.c17 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c16 {
  min-height: 600px;
}

.c18 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

.c20 {
  background: url(..//loader.gif) center center no-repeat;
}

.c19 {
  font-size: 16px;
  margin: 20px 0;
  color: #9e9577;
  font-weight: 600;
}

.c21 {
  text-align: center;
}

.c21 a {
  cursor: pointer;
  color: inherit;
}

.c21 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (max-width:640px) {
  .c4 {
    margin-top: 110px;
  }
}

@media screen and (max-width:640px) {
  .c12 {
    font-size: 14px;
    border: none;
    padding: 0;
    margin: 0 10px;
  }
}

@media screen and (max-width:640px) {
  .c13 {
    font-size: 14px;
  }
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
        data-testid="show-alert"
        id="show-alert"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.29 3.86L1.82 18a2 2 0 0 0 1.71 3h16.94a2 2 0 0 0 1.71-3L13.71 3.86a2 2 0 0 0-3.42 0z"
          />
          <line
            x1="12"
            x2="12"
            y1="9"
            y2="13"
          />
          <line
            x1="12"
            x2="12.01"
            y1="17"
            y2="17"
          />
        </svg>
      </div>
      <div
        class="c2"
      >
        Signaler un accident ou problème sur l'anneau
        <a
          class="c3"
          data-testid="notice-alert"
          href="https://docs.google.com/forms/d/e/1FAIpQLSdmrvUuZ953-3P8NVmNjiG8N78hY1lcdg_hVbr8lnYEqSjM9A/viewform?vc=0&c=0&w=1"
          id="notice-alert"
          rel="noopener noreferrer"
          target="_blank"
          title="Formulaire"
        >
          Formulaire
        </a>
      </div>
    </div>
    <main
      class="c4"
    >
      <header
        class="c5"
        id="header"
      >
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <a
              class="c8"
              data-testid="title-link"
              href="/"
              id="logo-link"
            >
              <img
                alt="logo"
                class="c9"
                src="/logo.png"
              />
            </a>
          </div>
          <div
            class="c10"
          >
            <nav>
              <ul
                class="c11"
              >
                <li>
                  <a
                    class="c12"
                    href="/blog"
                    id="blog-link"
                  >
                    Actualités
                  </a>
                </li>
              </ul>
            </nav>
            <a
              class="c13"
              href="/join"
              id="join-link"
            >
              Adhérer gratuitement
            </a>
            <div
              class="c14"
            >
              <button
                class="c15"
                data-testid="select-en"
                href="/"
                id="select-en"
              >
                EN
              </button>
            </div>
          </div>
        </div>
      </header>
      <section
        class="c16"
      >
        <div
          class="c17"
        >
          <h1
            class="c18"
            data-testid="h1-title"
          >
            Nous rejoindre
          </h1>
          <p
            class="c19"
          >
            Adhérer gratuitement à Vélo Longchamp :
          </p>
          <div
            class="c20"
          >
            <iframe
              data-hj-allow-iframe=""
              id="haWidget"
              scrolling="auto"
              src="https://www.helloasso.com/associations/velo-longchamp/adhesions/adhesion-association-velo-longchamp/widget"
              style="width: 100%; height: 750px;"
              title="Nous rejoindre"
            />
          </div>
          <div
            class="c21"
          >
            Si vous rencontrez un problème avec le formulaire, vous pouvez y accéder directement depuis le site de HelloAsso en cliquant 
            <a
              href="https://www.helloasso.com/associations/velo-longchamp/adhesions/adhesion-association-velo-longchamp"
              rel="noopener noreferrer"
              target="_blank"
              title="HelloAsso"
            >
              ici
            </a>
          </div>
        </div>
      </section>
    </main>
    <footer
      class="c22"
    >
      <p>
        ©Copyright 2019
         - 2022
         - Vélo Longchamp -
         
        Tous droits réservés
      </p>
      <p>
        -
      </p>
      <p
        id="legal-link"
      >
        <a
          href="/legal"
        >
          Mentions légales
        </a>
      </p>
    </footer>
    <div
      class="c23"
    >
      <p>
        En poursuivant votre navigation sur notre site, vous acceptez l'utilisation de cookies afin de nous permettre d'améliorer votre expérience utilisateur.
      </p>
      <div>
        <button
          data-testid="accept-cookie"
          id="accept-cookie"
        >
          OK
        </button>
        <a
          href="/legal"
        >
          En savoir plus
        </a>
      </div>
    </div>
  </div>
</div>
`;
